src/app/forms/custom-sections/adr/adr.component.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  Component, EventEmitter, Input, OnInit, Output,
+} from '@angular/core';
+import { TechRecordType } from '@dvsa/cvs-type-definitions/types/v3/tech-record/tech-record-vehicle-type';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormGroup } from '@forms/services/dynamic-form.types';
+import { AdrTemplate } from '@forms/templates/general/adr.template';
+
+@Component({
+  selector: 'app-adr',
+  templateUrl: './adr.component.html',
+  styleUrls: ['./adr.component.scss'],
+})
+export class AdrComponent implements OnInit {
+  @Input() techRecord!: TechRecordType<'hgv'> | TechRecordType<'trl'> | TechRecordType<'lgv'>;
+  @Input() isEditing = false;
+  @Input() disableLoadOptions = false;
+  @Output() formChange = new EventEmitter();
+
+  public template = AdrTemplate;
+  public form!: CustomFormGroup;
+  public adrDetails = [
```

Comment:
I worry about this, we now basically have some magic strings to check in a component, can this maybe be driven by the templates like the form is? The init creates the form, so you have the controls created etc

Reply:
Hmm think it would maybe need to be the template not the form

Reply:
Yeah, this is a symptom of the way we're handling the migrated data - the old adr data doesn't have the dangerous goods field, and all the other properties lie flat on the object. We need the field as this determines whether the dropdown is open, and when amending, if other adr fields should be nulled. I raised this with approval types on a previous ticket, because it seems weird to be that we'd do this kind of data correction on the front-end and not during the migration of data, especially it doesn't require any user input.

Not sure how this kind of logic could be implemented solely by the templates though...